api/types/role.go

```diff
@@ -1169,6 +1203,18 @@ func (r *RoleV6) CheckAndSetDefaults() error {
 			return trace.BadParameter("failed to process 'deny' db_permission #%v: %v", i+1, err)
 		}
 	}
+	for i := range r.Spec.Allow.SPIFFE {
+		err := r.Spec.Allow.SPIFFE[i].CheckAndSetDefaults()
+		if err != nil {
+			return trace.Wrap(err, "validating spec.allow.spiffe[%d]", i)
```

Comment:
Hm - I don't have too many strong opinions here. It's honestly nice to see more errors being wrapped in a way that's actually understandable compared to much of the older validation code which spits out something extremely unclear. One thing I do like to avoid is words like "failed" when wrapping errors, if every layer wraps the error in this way then you just end up repeating "failed" over and over again.

I'm happy to switch to 1-based indexing - but I feel like changing this here is likely a drop in the ocean without a better standard for error messaging/wrapping especially in regards to validation across the whole codebase.

Reply:
Yeah, I agree.
It was part of the reviewers of the PR that added the `db_permission` and it just seem odd to have two formats of reporting an error in a given item.

If we want to be consistent here, we have a lot to do :sweat_smile: 

As for the 1/0-based indexing, it's fine either way :+1: